applications/ShapeOptimizationApplication/python_scripts/algorithm_bead_optimization.py

```diff
@@ -112,6 +119,17 @@ def InitializeOptimizationLoop(self):
         self.mapper = mapper_factory.CreateMapper(self.design_surface, self.design_surface, self.mapper_settings)
         self.mapper.Initialize()
 
+        if self.filter_penalty_term:
+            penalty_filter_radius = self.algorithm_settings["penalty_filter_radius"].GetDouble()
+            filter_radius = self.mapper_settings["filter_radius"].GetDouble()
+            if abs(filter_radius - penalty_filter_radius) < 1e-9:
+                penalty_filter_settings = self.mapper_settings.Clone()
+                penalty_filter_settings["filter_radius"].SetDouble(self.algorithm_settings["penalty_filter_radius"].GetDouble())
```

Comment:
I think this is not necessary, as above you check whether the radii numerically match.

Reply:
Uh, thanks that was wrong. I mixed up the if/else conditions in my last commit.

Reply:
Fixed it.